sev-types/src/command/mod.rs

```diff
@@ -0,0 +1,178 @@
+    pub flags: PlatformStatusFlags,
+
+    /// Contains configuration information about the platform.
+    pub config: PlatformStatusConfig,
+
+    /// The number of valid guests maintained by the SEV firmware.
+    pub guest_count: u32,
+}
+
+/// Generate a new Platform Encryption Key (PEK).
+///
+/// (Chapter 5.7)
+#[repr(C, packed)]
+pub struct PekGenerate;
+
+/// Take ownership of the platform.
+///
+/// (Chapter 5.8)
+#[repr(C, packed)]
+pub struct PekCertificateSigningRequest<'a> {
+    /// The system's physical address of a memory region that
+    /// the platform will write the PEK certificate to.
+    addr: u64,
+
+    /// The length of the memory region (in bytes) that will contain
+    /// the PEK certificate.
+    len: u32,
+    _phantom: PhantomData<&'a ()>,
+}
+
+impl<'a> PekCertificateSigningRequest<'a> {
+    #![allow(missing_docs)]
+    pub fn new(addr: u64, len: u32) -> Self {
+        Self {
+            addr,
+            len,
+            _phantom: PhantomData,
+        }
+    }
+}
+
+/// Import the PEK and the OCA into the platform.
+///
+/// (Chapter 5.9; Table 22)
+#[repr(C, packed)]
+pub struct PekCertificateImport<'a> {
+    /// The system physical address of the memory region
+    /// that contains the PEK certificate.
+    pek_addr: u64,
+
+    /// The length of the memory region (in bytes) that
+    /// contains the PEK certificate.
+    pek_size: u32,
+
+    /// The system physical address of the memory region
+    /// that contains the OCA certificate.
+    oca_addr: u64,
+
+    /// The length of the memory region (in bytes) that contains
+    /// the OCA certificate.
+    oca_size: u32,
+    _phantom: PhantomData<&'a ()>,
+}
+
+impl<'a> PekCertificateImport<'a> {
+    #![allow(missing_docs)]
+    pub fn new(pek_addr: u64, pek_size: u32, oca_addr: u64, oca_size: u32) -> Self {
+        Self {
+            pek_addr,
+            pek_size,
+            oca_addr,
+            oca_size,
+            _phantom: PhantomData,
+        }
+    }
+}
+
+/// (Re)generate the Platform Diffie-Hellman (PDH).
+///
+/// (Chapter 5.10)
+#[repr(C, packed)]
+pub struct PdhGenerate;
+
+/// Retrieve the PDH and the certificate chain that testifies
+/// to the identity of the platform.
+///
+/// (Chapter 5.11; Table 25)
+#[repr(C, packed)]
+pub struct PdhCertificateExport<'a> {
+    pdh_addr: u64,
+    pdh_size: u32,
+    chain_addr: u64,
+    chain_size: u32,
+    _phantom: PhantomData<&'a ()>,
+}
+
+impl<'a> PdhCertificateExport<'a> {
+    #![allow(missing_docs)]
+    pub fn new(pdh_addr: u64, pdh_size: u32, chain_addr: u64, chain_size: u32) -> Self {
+        Self {
+            pdh_addr,
+            pdh_size,
+            chain_addr,
+            chain_size,
+            _phantom: PhantomData,
+        }
+    }
+}
+
+/// Request the CPU's unique identifier. Useful for obtaining
+/// a certificate for the CEK public key.
+///
+/// (Chapter 5.13; Table 30)
+#[repr(C, packed)]
+pub struct GetIdentifier(pub [u8; 64], pub [u8; 64]);
```

Comment:
I'm actually on the fence about this one.

Reply:
I put them in `sev-types` because these structures are described in the AMD SEV API spec and consumed by the firmware as they're laid out here. My goal is that the `sev-types` crate is an orthogonal package that anyone who wants to interact with SEV on a low-level can use. The only Linux-specific values here are the enums integral values that identify what ioctl action the kernel should take and those are in `iocuddle-sev`.